traits/tests/check_observe_timing.py

```diff
@@ -0,0 +1,568 @@
+        time to instantiate it, the time to reassign child, and the time to
+        reassign child.name
+    """
+    construct_parent_with_property = \
+        PARENT_WITH_PROPERTY_CONSTRUCTION_TEMPLATE.format(
+            property_args,
+            cached_property
+        )
+    construction_time = timeit.timeit(
+        stmt=construct_parent_with_property,
+        setup=CONSTRUCT_PARENT_SETUP,
+        number=N
+    )
+
+    instantiation_time = timeit.timeit(
+        stmt=INSTANTIATE_PARENT,
+        setup=CONSTRUCT_PARENT_SETUP + construct_parent_with_property,
+        number=N
+    )
+
+    reassign_child_time = timeit.timeit(
+        stmt=REASSIGN_CHILD,
+        setup=(
+            CONSTRUCT_PARENT_SETUP
+            + construct_parent_with_property
+            + INSTANTIATE_PARENT
+        ),
+        number=N
+    )
+
+    reassign_child_name_time = timeit.timeit(
+        stmt=REASSIGN_CHILD_NAME,
+        setup=(
+            CONSTRUCT_PARENT_SETUP
+            + construct_parent_with_property
+            + INSTANTIATE_PARENT
+        ),
+        number=N
+    )
+
+    return (
+        construction_time,
+        instantiation_time,
+        reassign_child_time,
+        reassign_child_name_time
+    )
+
+
+def report(description, benchmark_template, get_time, get_time_args):
+    """ Prints a readable benchmark report.
+
+    Parameters
+    ----------
+    descritption : str
+        The description of the benchmarking scenario being reported.
+    benchmark_template : str
+        The format string used to print the times for the benchmark in a clean,
+        formatted way
+    get_time : function
+        The function used to get the benchmark times for the current benchmark
+        scenario
+    get_time_args : list of tuples
+        The list of tuples containing the arguments to be passed the the
+        get_time function.  Note the first argument should give specifics about
+        the case being timed and will be printed as part of the report.  e.g.
+        ("@observe('name')"), or ("depends_on='name'", '@cached_property').  A
+        first argument of '' will be printed in the report as 'control'. The
+        first tuple should correspond to the control case.
+    """
+
+    print(description)
+
+    control_times = get_time(*get_time_args[0])
+
+    for args in get_time_args:
+        if args[0] == '':
+            print("control -")
+            case_times = control_times
+        else:
+            print(args[0] + ' -')
+            case_times = get_time(*args)
+
+        relative_times = map(
+            lambda times: times[1] / times[0], zip(control_times, case_times)
+        )
+
+        print(benchmark_template.format(*case_times, *relative_times))
+
+    print('-' * 80)
+
+
+def main():
+    BENCHMARK_TEMPLATE_3TIMES = ("(1): {0:.5f} {3:>6.2f} "
+                                 "(2): {1:.5f} {4:>6.2f} "
+                                 "(3): {2:.5f} {5:<6.2f}")
```

Comment:
Nit: Is this `{5:<6.2f}` instead of `{5:>6.2f}` (more consistent with others) deliberate?

Reply:
oops, nope it was a typo